kubejs/server_scripts/1000mek_wind_gene.js

```diff
@@ -1,100 +1,37 @@
-        'WSW',
-        'EWE'
-    ],{
-        S: 'mekanism:basic_control_circuit',
-        W: 'compactmekanismmachines:compressed_wind_generator_x32',
-        E: `mekanism:fuelwood_heater`
-    })
-
-    event.shaped('compactmekanismmachines:compressed_wind_generator_x512', [
-        'EWE',
-        'WEW',
-        'EWE'
-    ],{
-        E: 'mekanism:module_base',
-        W: 'compactmekanismmachines:compressed_wind_generator_x128',
+    let multipliers = [2, 8, 32, 128, 512, 2048, 8192, 32768, 131072, 532480]
+    multipliers.forEach((multiplier) => {
+        event.remove({ id: `compactmekanismmachines:compressed_wind_generator_x${multiplier}` })
     })
 
-    event.shaped('compactmekanismmachines:compressed_wind_generator_x2048', [
-        'SWS',
-        'WDW',
-        'SWS'
-    ],{
-        S: 'mekanism:seismic_vibrator',
-        W: 'compactmekanismmachines:compressed_wind_generator_x512',
-        D: 'mekanism:dynamic_valve'
-    })
-
-    event.shaped('compactmekanismmachines:compressed_wind_generator_x8192', [
-        'MWM',
-        'WGW',
-        'MWM'
-    ],{
-        M: 'mekanism:thermal_evaporation_valve',
-        W: 'compactmekanismmachines:compressed_wind_generator_x2048',
-        G: `megacells:cell_component_1m`
-    })
-
-    event.shaped('compactmekanismmachines:compressed_wind_generator_x32768', [
-        'HWH',
-        'WGW',
-        'HWH'
-    ],{
-        H: 'mekanism:sps_casing',
-        W: 'compactmekanismmachines:compressed_wind_generator_x8192',
-        G: `megacells:cell_component_4m`
-    })
-
-    event.shaped('compactmekanismmachines:compressed_wind_generator_x131072', [
-        ' W ',
-        'WSW',
-        ' W '
-    ],{
-        W: 'compactmekanismmachines:compressed_wind_generator_x32768',
-        S: 'mekaevolution:absolute_energy_cube'
-    })
+    event.recipes.mekanismMetallurgicInfusing(`compactmekanismmachines:compressed_wind_generator_x2`,'2x mekanismgenerators:wind_generator', '100x mekanism:redstone')
 
-    event.shaped('compactmekanismmachines:compressed_wind_generator_x532480', [
-        ' W ',
-        'WSW',
-        ' W '
-    ],{
-        W: 'compactmekanismmachines:compressed_wind_generator_x131072',
-        S: 'compactmekanismmachines:compressed_wind_generator_x8192'
-    })
+    function addWindGeneratorRecipe(result_multipliers,item1,item2){
+        event.shaped(`compactmekanismmachines:compressed_wind_generator_x${result_multipliers}`, item2==undefined?[
+            ' W ',
+            'WHW',
+            ' W '
+        ]:[
+            'SWS',
+            'WHW',
+            'SWS'
+        ],(item2==undefined)?{
+            H: item1,
+            W: `compactmekanismmachines:compressed_wind_generator_x${result_multipliers!=532480?result_multipliers/4:131072}`,
+        }:{
+            H: item1,
+            W: `compactmekanismmachines:compressed_wind_generator_x${result_multipliers!=532480?result_multipliers/4:131072}`,
+            S: item2
+        })
+    }
```

Comment:
無理に三項演算子を使わず中間の変数を用意することで、式のコピペをなくして構造を整理します。
```suggestion
    function addWindGeneratorRecipe(result_multipliers, item1, item2) {
        const recipeId = `compactmekanismmachines:compressed_wind_generator_x${result_multipliers}`
        const windGenPreviousMultiplier = multipliers[multipliers.indexOf(result_multipliers) - 1]

        const shape = []
        const ingredients = {
            H: item1,
            W: `compactmekanismmachines:compressed_wind_generator_x${windGenPreviousMultiplier}`
        }
        if (item2 == undefined) {
            shape.push(
                ' W ',
                'WHW',
                ' W '
            )
        } else {
            shape.push(
                'SWS',
                'WHW',
                'SWS'
            )
            ingredients["S"] = item2
        }

        event.shaped(recipeId, shape, ingredients)
    }
```